Merge module algorithm parts

mrouie opened the pull request:
Merged ModuleInFrame into a ModuleLocaiton.isInFrame

mrouie committed:
Create ModuleLocation.isInFrame from ModuleInFrame

In location.py, to avoid algorithmic recalculations, create a function
to determine if the module is in the fr`ame to be able to reuse
calculations associated with it in other functions

In location.py, update parameters for Hough Circle Detection function
call in ModuleLocation._circleDetection to match ModuleInFrame

In location.py, add typing where absent

In in_frame.py, replace error raised if too many circles are detected with
a False return value to avoid the need to handle errors in pipeline

mrouie committed:
Files reformatted

fallscameron01 commented on vision/module/location.py:
```diff
@@ -34,9 +34,37 @@ def __init__(self):
     ## Finding Distance to Module
 
-    def getDistance(self):
+    def getDistance(self) -> int:
```
The getDistance function can be removed from location. A more robust calculation is performed by get_module_depth.py

fallscameron01 commented on vision/module/location.py:
```diff
@@ -285,7 +313,7 @@ def _increaseBrightness(self, increase):
     ## Input Functions
 
-    def setImg(self, color, depth):
+    def setImg(self, color, depth) -> None:
```
> color, depth

Make sure you type function parameters as well. Should be:
color: np.ndarray, depth: np.ndarray

fallscameron01 commented on vision/module/location.py:
```diff
@@ -266,7 +294,7 @@ def _filterDepth(self):
         self.img = np.where(tempDepth < DEPTH_THRESH, self.img, 0)
 
-    def _increaseBrightness(self, increase):
+    def _increaseBrightness(self, increase) -> None:
```
Type parameter here as well.

fallscameron01 commented on vision/module/location.py:
```diff
@@ -49,7 +77,7 @@ def getDistance(self):
     ## Finding the Center
 
-    def getCenter(self):
+    def getCenter(self) -> tuple:
```
You should add a parameter here that allows us to use previous calculations if in_frame returned true. For example, you could have a bool parameter that defaults to true called recalculate. Then you could check whether recalculate is true before getCenter() runs _getSlopes() and _groupSlopes() again.

fallscameron01 reviewed (changes requested):
Looks like you adapted in_frame well. The main thing is that the point of this is to avoid recalculation, so adding a bool parameter like I described to getCenter() would help.

mrouie committed:
Remove extra functions and check for recalculation

- Create efficiency bool to determine if recalculation of hough circles,
  slopes, and grouping are needed.
- Remove getDistance since it is now done in get_module_depth.py
- Correct typing in setImg, _increaseBrightness, and getCenter

mrouie committed:
Reformat file

fallscameron01 commented on vision/module/location.py:
```diff
@@ -34,22 +34,45 @@ def __init__(self):
+            self._circleDetection()
+
+        if (
+            self.circles is None or self.circles.shape[0] > MAX_CIRCLES
```
> self.circles is None

This should be changed from is None to a constant called MIN_CIRCLES that would be set to 4 since there is also no point to calculations if there is 1, 2, or 3 circles.

fallscameron01 reviewed (changes requested):
Just a small change here and the PR good to go.

mrouie committed:
Updated minimum requirement for calculations

Minimum requirement is now 4 circles instead of no circles

mrouie committed:
Reformat files

mrouie committed:
Updated minimum requirement for calculations

mrouie committed:
Reformat file

fallscameron01 commented on vision/module/location.py:
```diff
@@ -34,22 +34,46 @@ def __init__(self):
+            self._circleDetection()
+
+        if (
+            np.shape(self.circles) < MIN_CIRCLES or np.shape(self.circles) > MAX_CIRCLES
```
You have to index np.shape(self.circles) since self.circles is a two-dimensional array, otherwise you would be comparing a tuple to an int. should be:

`np.shape(self.circles)[0] < MIN_CIRCLES or np.shape(self.circles)[0] > MAX_CIRCLES`

fallscameron01 commented on vision/module/in_frame.py:
```diff
@@ -50,11 +47,20 @@ def ModuleInFrame(color_image: np.ndarray) -> bool:
+        maxRadius=50,
+    )
+    if (
+        np.shape(self.circles) < MIN_CIRCLES or np.shape(self.circles) > MAX_CIRCLES
```
Same thing here.

fallscameron01 reviewed (changes requested):
You need to index np.shape since it will return a 2-tuple for the 2d array self.circles

mrouie committed:
Added min circles requirement and fixed errors

mrouie committed:
Reformat file

fallscameron01 reviewed (approved):
Looks good.
